101-sqlmi-new-vnet/azuredeploy.json

```diff
@@ -0,0 +1,170 @@
+{
+    "$schema": "http://schema.management.azure.com/schemas/2014-04-01-preview/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.1",
+    "parameters": {
+        "managedInstanceName": {
+            "type": "string",
+            "metadata": {
+                "description": "Enter managed instance name."
+            }
+        },
+        "administratorLogin": {
+            "type": "string",
+            "defaultValue": "cloudSA",
+            "metadata": {
+                "description": "Enter user name."
+            }
+        },
+        "administratorLoginPassword": {
+            "type": "securestring",
+            "metadata": {
+                "description": "Enter password."
+            }
+        },
+        "location": {
+            "type": "string",
+            "defaultValue": "",
+            "metadata": {
+                "description": "Enter location. If you leave this field blank resource group location would be used."
+            }
+        },
+        "virtualNetworkName": {
+            "type": "string",
+            "defaultValue": "",
```

Comment:
use a valid/non-empty defaultValue

Reply:
Is there a way to leave user a choice to fill in the name or use arm function to generate function instead? ARM functions don't pass validation for this field on the portal?

Reply:
You can just add a defaultValue, e.g.

"defaultValue": "virtualNetwork"

Reply:
changed to SQLMI-VNET fixed value